src/SculpinTools/AddLinksToId.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace  SculpinTools;
+
+use Couchbase\DocumentFragment;
+use Masterminds\HTML5;
+use Sculpin\Core\Event\ConvertEvent;
+use Sculpin\Core\Sculpin;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+final class AddLinksToId implements EventSubscriberInterface {
+
+
+    /**
+     * Returns an array of event names this subscriber wants to listen to.
+     *
+     * The array keys are event names and the value can be:
+     *
+     *  * The method name to call (priority defaults to 0)
+     *  * An array composed of the method name to call and the priority
+     *  * An array of arrays composed of the method names to call and respective
+     *    priorities, or 0 if unset
+     *
+     * For instance:
+     *
+     *  * array('eventName' => 'methodName')
+     *  * array('eventName' => array('methodName', $priority))
+     *  * array('eventName' => array(array('methodName1', $priority), array('methodName2')))
+     *
+     * @return array The event names to listen to
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            Sculpin::EVENT_AFTER_CONVERT => ['addLinksToIdTags']
+        ];
+    }
+
+    /**
+     * Adds a <a href="#id"> link to all h* elements with an id
+     *
+     * @param ConvertEvent $event
+     */
+    public function addLinksToIdTags(ConvertEvent $event) {
+
+        // only do it on markdown
+        if ($event->converter() != 'markdown') {
+            return;
+        }
+        $content = $event->source()->content();
+
+        if ( !empty($content)) {
+            $html5 = new HTML5();
+            $dom = $html5->loadHTML($content);
+            $xpath = new \DOMXPath($dom);
+            // search for all elements with an id attribute
+            $nodes = $xpath->query("//*[@id]");
+            foreach ($nodes as $node) {
+                /** @var \DOMElement $node */
+                // only do h1 - h6
+                switch ($node->nodeName) {
```

Comment:
I would rather have a constant of nodes you do this for. like ... = ['h1', 'h2'... ] and then check if it's in the array. :)

Reply:
also fixed